homatransport/src/transport/VectioTransport.cc

```diff
@@ -266,19 +267,26 @@ VectioTransport::processReqPkt(HomaPkt* rxPkt)
     else{
         assert(false);
     }
+    int bytesToSend = inboundRxMsg->msgByteLen;
+    //create and send per-packet grants for the message
+    do{
+        uint32_t pktDataBytes = std::min(bytesToSend, this->grantSizeBytes);
+        HomaPkt* grntPkt = new HomaPkt();
+        GrantFields grantFields;
+        grantFields.grantBytes = pktDataBytes;
+        grntPkt->setSrcAddr(inboundRxMsg->destAddr);
+        grntPkt->setDestAddr(inboundRxMsg->srcAddr);
+        grntPkt->setMsgId(inboundRxMsg->msgIdAtSender);
+        // grntPkt->setPriority(bytesToSend); //TODO think about the priority for grntPkt
+        grntPkt->setPktType(PktType::GRANT);
+        grntPkt->setGrantFields(grantFields);
+        grntPkt->setByteLength(grntPkt->headerSize());
+        bytesToSend -= pktDataBytes;
+        assert(bytesToSend >= 0);
 
-    //create and send a grant message for the added message
-    HomaPkt* grntPkt = new HomaPkt();
-    grntPkt->setSrcAddr(inboundRxMsg->destAddr);
-    grntPkt->setDestAddr(inboundRxMsg->srcAddr);
-    grntPkt->setMsgId(inboundRxMsg->msgIdAtSender);
-    // grntPkt->setPriority(bytesToSend);//TODO think about priority for grntPkt
-    grntPkt->setPktType(PktType::GRANT);
-    // grntPkt->setUnschedFields(unschedFields);
-    // grntPkt->setByteLength(pktDataBytes + grntPkt->headerSize());
-
-    // Send the packet out
-    socket.sendTo(grntPkt, grntPkt->getDestAddr(), destPort);
+        // Send the packet out
+        socket.sendTo(grntPkt, grntPkt->getDestAddr(), destPort);
+    }while(bytesToSend > 0);
```

Comment:
space.

Reply:
Taken care of all required spacing in the next pull request to avoid far too many merge conflicts to handle.